### PR TITLE
incorporate ui with api

### PR DESCRIPTION
**Pull request by Walaa-Volidis**
incorporate ui with api

### CODE REVIEW

File: src/app/api/createTask/route.ts [hunk 1]
```diff
@@ -15,12 +15,13 @@ export async function POST(request: Request) {
       title: formData.get("title") as string,
```
Comment (nmaswood) [inline src/app/api/createTask/route.ts:15]: **Nit: for these file names I would stick the convention of kebab-case**

e.g. create-task vs createTask

File: src/app/components/TaskForm.tsx [hunk 2]
```diff
@@ -20,32 +20,34 @@ import { PlusCircle } from "lucide-react";
...
+    try {
+      await addTask(formData);
+      setTask({
+        title: "",
+        description: "",
+        category: "choose",
+        date: "",
+      });
+    } catch (error) {
+      console.error("Error submitting task:", error);
```
Comment (nmaswood) [inline src/app/components/TaskForm.tsx:48]: **Suggestion: use a toast to show the fact that the request failed to the user**

File: src/app/components/TaskForm.tsx [hunk 3]
```diff
@@ -20,32 +20,34 @@ import { PlusCircle } from "lucide-react";
...
-      date: "",
-    });
-  };
+  async function onSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const formData = new FormData(event.currentTarget);
+    formData.append("userId", userId);
+    try {
+      await addTask(formData);
+      setTask({
```
Comment (nmaswood) [inline src/app/components/TaskForm.tsx:41]: **Suggestion: can you reload the page or have addTask return all the userTasks from the server?**

File: src/app/hooks/useTasks.ts [hunk 4]
```diff
@@ -1,58 +1,57 @@
...
+      const data = await response.json();
+      setTasks(data);
+    } catch (error) {
+      console.error("Failed to fetch tasks:", error);
+    }
   };
 
-  const deleteTask = (id: string) => {
-    setTasks((prev) => prev.filter((task) => task.id !== id));
+  const addTask = async (formData: FormData) => {
```
Comment (nmaswood) [inline src/app/hooks/useTasks.ts:26]: **Suggestion: can you experiment with useing use-swr and use-mutate-swr for these requests?**

It saves you a little bit of boiler plate and can make things easier going forward

**Review by nmaswood — APPROVED** [review]

File: src/app/components/TaskForm.tsx (hunk 2 above)
Comment (Walaa-Volidis) [reply src/app/components/TaskForm.tsx:48]: done

File: src/app/api/createTask/route.ts (hunk 1 above)
Comment (Walaa-Volidis) [reply src/app/api/createTask/route.ts:15]: done

**Pull request merged**